app/controllers/articles_controller.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+class ArticlesController < ApplicationController
+  before_action :set_article, only: %i[show edit update destroy]
+
+  def index
+    @articles = Article.sorted.paginate(page: params[:page])
+  end
+
+  def show; end
+
+  def new
+    @article = Article.new
+  end
+
+  def create
+    @article = Article.new(article_params)
+    if @article.save
+      redirect_to @article, notice: I18n.t('controllers.admin.article.create_success')
+    else
+      flash.now[:alert] = I18n.t('controllers.admin.article.create_unsuccess')
+      render :new
+    end
+  end
+
+  def edit; end
+
+  def update
+    if @article.update(article_params)
+      redirect_to article_path(@article), notice: I18n.t('controllers.admin.article.create_unsuccess')
+    else
+      render :edit, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    article = set_article
+    if @article.destroy!
```

Comment:
add memoized method as i said before and then just do:  "if article.destroy"

btw shoud here be destroy with "!"?

Reply:
guys and girls, forget about memoization for now, you are making MUCH important mistakes

START doing pair programming if you want to finish academy successfully
10 PRs are waiting!

Reply:
and why I said stop worry about memoization because what you suggested is wrong, it's not needed there
it's a simple action. 1) find object 2) destroy 3) redirect to another page